Add initial production release of Cloudsmith integration (v1.0.0)

opswithranjan opened the pull request:
🧩 Add Cloudsmith Integration to Datadog (v1.0.0)

This PR introduces the initial production release (v1.0.0) of the Cloudsmith integration for Datadog Agent.

✅ Features Implemented:
• Quota Usage Metrics
• cloudsmith.storage_used (percentage)
• cloudsmith.bandwidth_used (percentage)
• cloudsmith.storage_used_gb, cloudsmith.bandwidth_used_gb
• cloudsmith.storage_used_bytes, cloudsmith.bandwidth_used_bytes
• Plan limits included for both storage and bandwidth
• Entitlement Token Metrics
• cloudsmith.token_count
• cloudsmith.token_bandwidth_total
• cloudsmith.token_download_total
• Security and Policy Events
• vulnerabilities event
• vulnerability_policy_violation event
• license_policy_violation event
• Audit Logs
• audit_log events with actor + action context
• Organisation Members
• cloudsmith.member.active (per user)
• org_members summary event with 2FA and last login insights

🛠️ Other Notes:
• Handles pagination where required (e.g. policy violations, members).
• Deduplication logic on members list based on user slug, not display name.
• Converts timestamps to epoch for proper Datadog ingestion.

🙋 Maintainer Request:

I’d like to be listed as a maintainer for this integration and actively maintain/iterate it.

dd-dominic commented:
@opswithranjan can you confirm that this is an update to the current integration? If so, please update the PR title appropriately.

opswithranjan commented:
Hey @dd-dominic , this is an update to the existing Cloudsmith integration. The earlier releases were v0.0.1 and v0.0.2, and this PR rolls out the first production-ready release as v1.0.0. I’ll update the PR title accordingly to reflect this.

dd-dominic commented:
> Hey @dd-dominic , this is an update to the existing Cloudsmith integration. The earlier releases were v0.0.1 and v0.0.2, and this PR rolls out the first production-ready release as v1.0.0. I have updated the PR title accordingly to reflect this.

@opswithranjan Can you clarify what you mean by production-ready? The Cloudsmith integration has been live since late 2021.

Additionally, please be sure to fix the failed validations before we can continue the review.

opswithranjan commented:
Thanks for the review!

By production-ready, I meant this release now follows Semantic Versioning (SemVer)—shifting from 0.x.x (commonly used for early or internal development) to v1.0.0, which signifies a stable and maintained version.

While the Cloudsmith integration was originally released in 2021, it had gone stale after the initial versions. With this release, we’re reviving it with the intention to actively maintain and expand it going forward.

There were some minor validation issues due to a black version mismatch—I’ll address those shortly

dd-dominic commented on cloudsmith/CHANGELOG.md:
```diff
@@ -1,2 +1,29 @@
 # CHANGELOG - Cloudsmith
 
+### 1.0.0 / 2025-05-01
+
+### Added
+* Initial production release of the Cloudsmith Datadog integration.
+* Metrics for:
+  * Storage usage:
+    * `cloudsmith.storage_used`: Storage usage percentage.
+    * `cloudsmith.storage_used_bytes`: Raw storage used in bytes.
+    * `cloudsmith.storage_peak_bytes`: Peak storage usage in bytes.
+    * `cloudsmith.storage_configured_bytes`: Configured storage capacity in bytes.
+  * Bandwidth usage:
+    * `cloudsmith.bandwidth_used`: Bandwidth usage percentage.
+    * `cloudsmith.bandwidth_used_bytes`: Raw bandwidth used in bytes.
+    * `cloudsmith.bandwidth_configured_bytes`: Configured bandwidth capacity in bytes.
+  * Token usage:
+    * `cloudsmith.token_count`: Number of entitlement tokens in the org.
+    * `cloudsmith.token_bandwidth_total`: Total bandwidth used by tokens.
+    * `cloudsmith.token_download_total`: Total downloads served via tokens.
+  * Member metrics:
+    * `cloudsmith.member.active`: Total active members.
+    * `cloudsmith.member.role.owner`, `.manager`, `.member`: Member counts by role.
+    * `cloudsmith.member.has_2fa`: Members with 2FA enabled.
+* Events:
+  * Security vulnerability scan results (`@aggregation_key:vulnerabilities`)
+  * Audit log activity (`@aggregation_key:audit_log`)
+  * Policy violations for vulnerabilities and license compliance
+* Support for `cloudsmith.members` data collection and Datadog event enrichment including detailed summaries and table-style logs.
```
Feel free to add back "Support for `cloudsmith.members` data collection and Datadog event enrichment including detailed summaries and table-style logs." but make it more clear. I'm not exactly sure what it means as is.

dd-dominic commented on cloudsmith/README.md:
```diff
@@ -36,22 +36,50 @@ For Agent v7.21+ / v6.21+, follow the instructions below to install the Cloudsmi
 ### Metrics
 
-See [metadata.csv][8] for a list of metrics provided by this check.
+This integration provides the following metrics:
+
+- `cloudsmith.storage.percentage_used`: Percentage of storage used.
+- `cloudsmith.storage.used_bytes`: Total storage used in bytes.
+- `cloudsmith.storage.peak_bytes`: Peak storage usage in bytes.
+- `cloudsmith.bandwidth.percentage_used`: Percentage of bandwidth used.
+- `cloudsmith.bandwidth.used_bytes`: Total bandwidth used in bytes.
+- `cloudsmith.members.total`: Total number of members in the organization.
+- `cloudsmith.members.owners`: Total number of owners.
+- `cloudsmith.members.managers`: Total number of managers.
+- `cloudsmith.members.readers`: Total number of readers.
+- `cloudsmith.tokens.total`: Number of entitlement tokens configured.
+- `cloudsmith.tokens.active`: Number of active entitlement tokens.
+- `cloudsmith.vulnerabilities.total`: Number of vulnerabilities found.
+- `cloudsmith.license_policy_violations.total`: Number of license policy violations detected.
+- `cloudsmith.vulnerability_policy_violations.total`: Number of vulnerability policy violations detected.
+
+See [metadata.csv][8] for the full list and detailed descriptions.
```
Revert these changes to keep original. Just be sure to list all metrics in the metadata file.
```suggestion
See [metadata.csv][8] for a list of metrics provided by this check.
```

dd-dominic commented on cloudsmith/README.md:
```diff
@@ -36,22 +36,50 @@ For Agent v7.21+ / v6.21+, follow the instructions below to install the Cloudsmi
 ### Events
 
-All Cloudsmith related events collected appear within the Datadog Event Stream with the `source:cloudsmith` property. They are collected every five minutes to reduce the number of requests sent to the Cloudsmith API.
+All Cloudsmith-related events collected appear within the Datadog Event Stream with the `source:cloudsmith` property. Events are collected every five minutes to reduce the number of requests sent to the Cloudsmith API.
```
```suggestion
All Cloudsmith-related events collected appear within the Datadog Event Explorer with the `source:cloudsmith` tag. Events are collected every five minutes to reduce the number of requests sent to the Cloudsmith API.
```

dd-dominic commented on cloudsmith/manifest.json:
Add classifier_tags for submitted data types (see docs here: https://docs.datadoghq.com/developers/integrations/check_references/#classifier-tags)

dd-dominic commented on cloudsmith/README.md:
```diff
@@ -36,22 +36,50 @@ For Agent v7.21+ / v6.21+, follow the instructions below to install the Cloudsmi
+They are accessible using the following aggregation keys:
 
-They are accessible with aggregation keys: `@aggregation_key:audit_log` and `@aggregation_key:vulnerabilities`.
+- `@aggregation_key:audit_log`
+- `@aggregation_key:vulnerabilities`
+- `@aggregation_key:vulnerability_policy_violation`
+- `@aggregation_key:license_policy_violation`
+- `@aggregation_key:org_members_summary`
+- `@aggregation_key:quota_summary`
```
Remove as this is no longer supported (https://docs.datadoghq.com/service_management/events/guides/migrating_to_new_events_features/#event-aggregations-are-no-longer-performed-or-displayed-in-the-ui)
```suggestion
```

dd-dominic commented on cloudsmith/README.md:
```diff
@@ -36,22 +36,50 @@ For Agent v7.21+ / v6.21+, follow the instructions below to install the Cloudsmi
 
 See [service_checks.json][9] for a list of service checks provided by this integration.
 
 ### Events
```
Currently, this ### Events section only shows in the public docs page. I recommend moving this section under the ## Overview section in order to display both in-app and in the public docs.

dd-dominic commented on cloudsmith/README.md:
```diff
@@ -36,22 +36,50 @@ For Agent v7.21+ / v6.21+, follow the instructions below to install the Cloudsmi
+- `@aggregation_key:org_members_summary`
+- `@aggregation_key:quota_summary`
 
 ## Troubleshooting
```
```suggestion
## Support
```

dd-dominic commented on cloudsmith/metadata.csv:
If metrics are being tagged, please add sample_tags column https://docs.datadoghq.com/developers/integrations/check_references/#metrics-metadata-file

dd-dominic commented on cloudsmith/manifest.json:
Make description more descriptive, such as "Collect metrics and events from Cloudsmith"

dd-dominic commented on cloudsmith/README.md:
Under ## Installation, replace "<INTEGRATION_VERSION>" with "1.0.0"

dd-dominic commented on cloudsmith/README.md:
```diff
@@ -3,7 +3,7 @@
 ## Overview
```
This overview needs to be rewritten to answer the following:

1. What does your company/platform/product do?
2. How does this integration enhance the customers’ workflows or observability?
3. What data is collected from your product, and which Datadog data types is it mapped to (e.g., logs, metrics, events)?

dd-dominic commented on cloudsmith/manifest.json:
Add dashboard image to media attribute (see example https://github.com/DataDog/integrations-extras/blob/master/ably/manifest.json#L14)

dd-dominic reviewed (changes requested):
Quite a few comments and suggestions since we've updated our requirements since this integration's initial release. 

In addition to the individual comments and suggestions, the new metrics and events should be captured in the prebuilt dashboard. Once dashboard updates are made, please also update the screenshot.
[Integration_Dashboard_Template.json](https://github.com/user-attachments/files/20072390/Integration_Dashboard_Template.json)
Please also add color grouping headers for better visual searation (see attached template for reference).

dd-dominic commented:
> Quite a few comments and suggestions since we've updated our requirements since this integration's initial release.
> 
> In addition to the individual comments and suggestions, the new metrics and events should be captured in the prebuilt dashboard. Once dashboard updates are made, please also update the screenshot. [Integration_Dashboard_Template.json](https://github.com/user-attachments/files/20072390/Integration_Dashboard_Template.json) Please also add color grouping headers for better visual searation (see attached template for reference).

@opswithranjan please address dashboard updates^ and fix the failed validations.

dd-dominic reviewed (changes requested):
Requesting additional changes. Since the logo was updated, this will take ~2 weeks until we can merge to production.

As for the failed validation, can you merge the latest master branch to see if that fixes the issue?

urseberry commented:
Flagging for the team that I’m the Documentation reviewer for this PR.

urseberry commented on cloudsmith/CHANGELOG.md:
```diff
@@ -1,2 +1,14 @@
+***Added***:
+
+* Metric support for storage usage, bandwidth usage, token usage, and member activity
+* Event support for security vulnerability scan results, audit logs, policy violations related to vulnerabilities and license compliance
```
```suggestion
* Event support for security vulnerability scan results, audit logs, policy violations related to vulnerabilities, and license compliance
```
By adding an Oxford comma, I'm assuming "policy violations related to vulnerabilities" and "license compliance" are two separate list items.

opswithranjan commented:
@dd-dominic I’ve implemented the requested changes and updated the dashboard screenshot accordingly. All test cases are now passing. Could you please review and approve the PR when you get a chance?

dd-dominic commented on cloudsmith/CHANGELOG.md:
```diff
@@ -1,2 +1,14 @@
 # CHANGELOG - Cloudsmith
 
+### 1.0.0 / 2025-05-01
```
```suggestion
### 1.0.0 / 2025-06-06
```

dd-dominic commented on cloudsmith/assets/dashboards/cloudsmith_overview.json:
Dashboard looks good. One additional request: remove "Cloudsmith Integration on Datadog" and "Cloudsmith Metadata CSV" links (this info will be surfaced on integration tile already).

opswithranjan replied:
Done

opswithranjan commented:
Hi @dd-dominic 
I have made further changes to the release date and removed those links. I also updated the dashboard screenshot after those changes.

dd-dominic commented:
@urseberry ready for review

dd-dominic commented:
> Thanks a lot @dd-dominic 🙏🏻 It’s live now! The only thing I noticed is that the images aren’t showing up in the integration listing here: [Cloudsmith](https://docs.datadoghq.com/integrations/cloudsmith/?site=us)

@opswithranjan the images are live now - just took a bit to be deployed
